xonsh/platform.py

```diff
@@ -0,0 +1,141 @@
+""" Module for platform-specific constants and implementations """
+
+import os
+import platform
+import sys
+
+try:
+    import distro
+except ImportError:
+    distro = None
+
+
+#
+# OS
+#
+
+ON_DARWIN = platform.system() == 'Darwin'
+ON_LINUX = platform.system() == 'Linux'
+ON_WINDOWS = platform.system() == 'Windows'
+
+ON_POSIX = (os.name == 'posix')
+
+
+
+#
+# Python & packages
+#
+
+PYTHON_VERSION_INFO = sys.version_info[:3]
+
+if PYTHON_VERSION_INFO < (3, 5, 0):
+    from pathlib import Path
+
+    class DirEntry(Path):
+        def inode(self, *args, **kwargs):
+            raise NotImplementedError
+
+    def scandir(path):
+        """ Compatibility wrapper for os.scandir from Python 3.5+ """
+        return (DirEntry(x) for x in os.listdir(path))
+else:
+    from os import scandir
+
+
+ON_ANACONDA = any(s in sys.version for s in ('Anaconda', 'Continuum'))
+
+try:
+    import pygments
+except ImportError:
+    HAS_PYGMENTS, PYGMENTS_VERSION = False, None
+else:
+    HAS_PYGMENTS, PYGMENTS_VERSION = True, pygments.__version__
+
+
+try:
+    import readline
+except ImportError:
+    HAS_READLINE = False
+else:
+    HAS_READLINE = False
+
+
+try:
+    import prompt_toolkit
+except ImportError:
+    HAS_PROMPT_TOOLKIT, PROMPT_TOOLKIT_VERSION = False, None
+    PROMPT_TOOLKIT_VERSION_INFO = None
+else:
+    HAS_PROMPT_TOOLKIT = True
+    PROMPT_TOOLKIT_VERSION = getattr(prompt_toolkit, '__version__', '<0.57')
+    PROMPT_TOOLKIT_VERSION_INFO = tuple(
+        int(x) for x in PROMPT_TOOLKIT_VERSION.strip('<>+-=.').split('.'))
+
+
+if HAS_PROMPT_TOOLKIT or ON_WINDOWS:
+    BEST_SHELL_TYPE = 'prompt_toolkit'
+else:
+    BEST_SHELL_TYPE = 'readline'
+
+
+#
+# Encoding
+#
+
+DEFAULT_ENCODING = sys.getdefaultencoding()
+
+
+#
+# Linux distro
+#
+
+if ON_LINUX:
+    if distro:
+        LINUX_DISTRO = distro.id()
+    elif PYTHON_VERSION_INFO < (3, 7, 0):
+        LINUX_DISTRO = platform.linux_distribution()
+    elif '-ARCH-' in platform.platform():
+        LINUX_DISTRO = 'arch'  # that's the only one we need to know for now
+    else:
+        LINUX_DISTRO = 'unknown'
+else:
+    LINUX_DISTRO = None
+
+
+#
+# Windows
+#
+
+try:
+    import win_unicode_console
+except ImportError:
+    win_unicode_console = None
+
+
+#
+# Bash completions defaults
+#
+
+if LINUX_DISTRO == 'arch':
+    BASH_COMPLETIONS_DEFAULT = \
+        ('/etc/bash_completion',
+         '/usr/share/bash-completion/completions')
+elif ON_LINUX:
+    BASH_COMPLETIONS_DEFAULT = \
+        ('/usr/share/bash-completion',
+         '/usr/share/bash-completion/completions')
+elif ON_DARWIN:
+    BASH_COMPLETIONS_DEFAULT = \
+        ('/usr/local/etc/bash_completion',
+         '/opt/local/etc/profile.d/bash_completion.sh')
+else:
+    BASH_COMPLETIONS_DEFAULT = \
+        ()
```

Comment:
This can be on the previous line. Also there is no reason to use backslashes here since you can just use and opening paren.